### PR TITLE
feat: Add MCP Client support for bidirectional MCP integration

### PR DESCRIPTION
**Pull request by Jameswlepage**
## Summary
- Add `McpClient` class for connecting to external MCP servers
- Extend `McpAdapter` to support client management alongside servers  
- Automatically register remote MCP tools/resources as WordPress abilities
- Support multiple authentication methods (Bearer, API key, Basic auth)
- Follow existing server patterns for consistency and maintainability

## Key Features
- **Bidirectional Integration**: WordPress can now both expose and consume MCP capabilities
- **Automatic Ability Registration**: Remote tools become local abilities with `mcp-{client-id}/` prefix
- **Multiple Transport Support**: SSE and standard MCP protocol support
- **Authentication**: Bearer token, API key, and basic auth support
- **Error Handling**: Consistent with existing server error handling patterns

## Test Plan
- [x] Unit tests for McpClient functionality
- [x] Integration tests with WordPress Domains MCP server  
- [x] Authentication method validation
- [x] Error handling and edge cases
- [x] Ability registration and execution

### CODE REVIEW

File: includes/Core/McpAdapter.php
Comment (justlevine) [inline includes/Core/McpAdapter.php:1]: Wondering for SRP/Maintainability if it makes sense to have a separate `ClientRegistry` from our `ServerRegistry`. related #26

File: includes/Core/McpClient.php [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+
+use WP\MCP\Infrastructure\ErrorHandling\Contracts\McpErrorHandlerInterface;
+use WP\MCP\Infrastructure\Observability\Contracts\McpObservabilityHandlerInterface;
+
+/**
+ * MCP Client for connecting to external MCP servers.
+ * 
+ * Mirrors the McpServer architecture for consistency.
+ */
+class McpClient {
```
Comment (justlevine) [inline includes/Core/McpClient.php:20]: Buys us time to shape the API. There's currently no way to overload this class anyway, and there's a significant number of public methods on this class.

```suggestion
final class McpClient {
```

File: includes/Core/McpClient.php [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+		// Auto-connect on construction
+		$this->connect();
+	}
+
+	/**
+	 * Connect to the MCP server.
+	 *
+	 * @return bool True on success, false on failure.
+	 */
+	public function connect(): bool {
```
Comment (justlevine) [inline includes/Core/McpClient.php:124]: Is this meant to be consumed publicly? e.g. ` if ( ! $my_client->is_connected() ) { $client->connect(); }`

If not, let's make the method `private` until we have a concrete use case to expose it `public`.

If so, I suggest we get it out of constructor and dogfood our own APIs by calling it in `MCPAdapter::create_client()`.

File: includes/Core/McpClient.php [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+				),
+				'clientInfo' => array(
+					'name'    => 'WordPress MCP Client',
+					'version' => '1.0.0',
+				),
+			) );
+
+			if ( is_wp_error( $response ) ) {
+				$error_msg = 'MCP Initialize Request Failed: ' . $response->get_error_message();
+				error_log( $error_msg );
```
Comment (justlevine) [inline includes/Core/McpClient.php:144]: I'm still only partially through my initial code review , but I believe these are handled by the `ErrorLogMcpErrorHandler` abstraction when `$this->error_handler->log() gets called

 - https://github.com/WordPress/mcp-adapter/blob/f41bf0e2c0ea94564a8d45a9bb412c1e33780bf4/includes/Infrastructure/ErrorHandling/ErrorLogMcpErrorHandler.php#L30-L40

File: includes/Core/McpAdapter.php [hunk 4]
```diff
@@ -285,4 +293,201 @@ public function get_server( string $server_id ): ?McpServer {
...
 	}
+
+	/**
+	 * Create and register a new MCP client.
+	 *
+	 * @param string      $client_id   Unique identifier for the client.
+	 * @param string      $server_url  MCP server URL to connect to.
+	 * @param array       $config      Client configuration (auth, timeouts, etc.).
+	 * @param string|null $error_handler The error handler class name.
+	 * @param string|null $observability_handler The observability handler class name.
```
Comment (justlevine) [inline includes/Core/McpAdapter.php:304]:
```suggestion
	 * @param ?class-string<\WP\MCP\Infrastructure\ErrorHandling\Contracts\McpErrorHandlerInterface>         $error_handler The error handler class name.
	 * @param ?class-string<\WP\MCP\Infrastructure\Observability\Contracts\McpObservabilityHandlerInterface> $observability_handler The observability handler class name.
```

File: includes/Core/McpAdapter.php [hunk 5]
```diff
@@ -285,4 +293,201 @@ public function get_server( string $server_id ): ?McpServer {
...
+	 * @return \WP\MCP\Core\McpClient|\WP_Error Client instance or error.
+	 * @throws \Exception If the client already exists or invalid handlers.
```
Comment (justlevine) [inline includes/Core/McpAdapter.php:306]: These are (thankfully) returned
```suggestion
```

**Review by justlevine — COMMENTED** [review]
Didn't dive too deeply to avoid being redundant to phpcs/phpstan/ #27

More generally, I think the code patterns here like using `WP_Error` are a marked improvement over the existing patterns in `McpServer` and its deps; if consistency is a concern, _those existing_ classes/methods should IMO be brought to align with what's here.

**Comment by Jameswlepage** [comment]
We should revisit this concept of adding "Client Support", but also, probably close the current PR as this is very out of date atm. I'd like to see if this can land _by_ 7.0, even if experimental.

**Comment by galatanovidiu** [comment]
Closing in favor of: https://github.com/Automattic/php-mcp-client
Mabe it will be merged into mcp-adpater
